zug/transducer/eager.hpp

```diff
@@ -22,6 +22,56 @@ namespace zug {
 struct eager_tag
 {};
 
+struct eager_t
+{
```

Comment:
I think this type can actually go, and be a normal function instead as it used to be...